src/backend/joanie/signature/backends/lex_persona.py

```diff
@@ -522,7 +524,20 @@ def delete_signing_procedure(self, reference_id: str):
         url = f"{base_url}/api/workflows/{reference_id}"
         headers = {"Authorization": f"Bearer {token}"}
 
-        response = requests.delete(url, headers=headers, timeout=timeout)
+        try:
+            response = requests.delete(url, headers=headers, timeout=timeout)
+        except ReadTimeout as exception:
+            logger.error(
+                exception,
+                extra={
+                    "context": {
+                        "signature_backend_reference": reference_id,
+                    }
+                },
+            )
+            raise BackendTimeout(
```

Comment:
Why not raising a DeleteSignatureProcedureFailed here ?
(Sorry for not thinking about this earlier)

Reply:
I think it's more accurate to distinct both cases :
* The workflow is being deleted but takes a while
* the workflow reference does not exist at the signature provider side